judge/views/problem_data.py

```diff
@@ -44,7 +44,19 @@ class ProblemDataForm(ModelForm):
     def clean_zipfile(self):
         if hasattr(self, 'zip_valid') and not self.zip_valid:
             raise ValidationError(_('Your zip file is invalid!'))
-        return self.cleaned_data['zipfile']
+
+        zipfile = self.cleaned_data['zipfile']
+        if zipfile and zipfile.name == 'init.yml':
```

Comment:
Rather than this very specific check, I feel like we should enforce that zip files... end with `.zip`.

Reply:
Fixed. The check for the generator has been left as is.